### PR TITLE
Add support for custom client hello specs

### PR DESCRIPTION
**Pull request by hwh33**
Part of https://github.com/getlantern/lantern-internal/issues/3349

### CODE REVIEW

**Comment by hwh33** [comment]
This package is used to dial HTTPS proxies.  If we want to use custom hellos (mimicking the browsers) for those handshakes, we will need something like this.

File: tlsdialer.go [hunk 1]
```diff
@@ -33,8 +34,14 @@ type Dialer struct {
...
 	Network        string
 	SendServerName bool
+
 	// Force validation of a specific name other than the SNI name or dialed hostname
-	ForceValidateName  string
+	ForceValidateName string
+
+	// Must be provided if ClientHelloID is set to tls.HelloCustom. This field is ignored if
+	// ClientHelloID is not set to tls.HelloCustom.
+	ClientHelloSpec *tls.ClientHelloSpec
```
Comment (oxtoacart) [inline tlsdialer.go:43]: To be super safe, maybe also check if ClientHelloSpec is specified but ClientHelloID is not HelloCustom?

Comment (hwh33) [reply tlsdialer.go:43]: Sure, I can do that

**Comment by hwh33** [comment]
I broke the tests somehow.  Kind of confused by that, but I'll figure it out.

**Comment by hwh33** [comment]
Okay, it's just the addition of another test that broke things.  Seems to be a timing-related failure with some of the global objects used by the tests.

**Comment by hwh33** [comment]
@oxtoacart I added another commit which fixes the tests.  Would you mind taking a look?  It depends on [a small PR to keyman](https://github.com/getlantern/keyman/pull/13).

**Comment by hwh33** [comment]
I'm going to go ahead and merge this.

**Pull request merged**